geode-core/src/main/java/org/apache/geode/management/internal/operation/OperationManager.java

```diff
@@ -14,75 +14,95 @@
  */
 package org.apache.geode.management.internal.operation;
 
+import java.util.Date;
 import java.util.Map;
 import java.util.UUID;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 import java.util.concurrent.ExecutorService;
-import java.util.function.Function;
+import java.util.function.BiFunction;
 
 import org.apache.geode.annotations.Experimental;
+import org.apache.geode.cache.Cache;
+import org.apache.geode.internal.cache.InternalCache;
 import org.apache.geode.internal.logging.LoggingExecutors;
 import org.apache.geode.management.api.ClusterManagementOperation;
 import org.apache.geode.management.api.JsonSerializable;
 import org.apache.geode.management.internal.operation.OperationHistoryManager.OperationInstance;
+import org.apache.geode.management.operation.RebalanceOperation;
 
 @Experimental
 public class OperationManager implements AutoCloseable {
-  private final Map<Class<? extends ClusterManagementOperation>, Function> performers;
+  private final Map<Class<? extends ClusterManagementOperation>, BiFunction> performers;
   private final OperationHistoryManager historyManager;
   private final Executor executor;
+  private final InternalCache cache;
 
-  public OperationManager(OperationHistoryManager historyManager) {
+  public OperationManager(InternalCache cache, OperationHistoryManager historyManager) {
+    this.cache = cache;
     this.historyManager = historyManager;
     this.executor = LoggingExecutors.newThreadOnEachExecute("CMSOpPerformer");
 
     // initialize the list of operation performers
     performers = new ConcurrentHashMap<>();
+    registerOperation(RebalanceOperation.class, RebalanceOperationPerformer::perform);
```

Comment:
if we eventually replace this with service loader, (you can see how commands are loaded by each extension to load all the custom commands), what needs to be put in the map is an instance of Performer, so I do believe we need to register a stateless performer instance in the map. It probably would help if we have a  dedicated interface for all the performers.

Reply:
The current: RebalanceOperationPerformer::perform is a stateless performer. I think you might be right that it could help if we had our own interface that defines this contract more explicitly. But this BiFunction one might work. Would it be okay to delay adding support for service loader until someone works on that story and then they could add the performer interface if needed?

Reply:
yeah, this can be delayed.